protocol/app/module_accounts_test.go

```diff
@@ -72,6 +74,9 @@ func TestMaccPerms(t *testing.T) {
 		"rewards_vester":         nil,
 		"community_treasury":     nil,
 		"community_vester":       nil,
+		// required for burn msg in wasmvm
+		// https://github.com/CosmWasm/wasmvm/blob/1c3fdc2a4402e527617ec72fe53f114b24899a01/types/msg.go#L116-L121
+		"wasm": {"burner"},
```

Comment:
What does this mean exactly? Can any contract dispatch `BurnMsg`, and what are the restrictions? Also this comment should not be in the test file but in the actual permissions

Reply:
Updated the comment. Went through the code and IIUC any contract can dispatch BurnMsg for itself https://github.com/CosmWasm/wasmd/blob/v0.51.0/x/wasm/keeper/handler_plugin.go#L225-L244

Reply:
Ok thanks. It looks like contract can only burn coins that belong to itself
`// Important if a contract controls significant token supply that must be retired.` what do you mean by this? Seems we can leave out this comment

Reply:
Ah... i added that to just say that if a contract has a lot of tokens and is retired, those tokens need to be burned